content/03_CreatingObservableSequences.md

```diff
@@ -0,0 +1,1274 @@
+
+```csharp
+static void StartAction()
+{
+    var start = Observable.Start(() =>
+        {
+            Console.Write("Working away");
+            for (int i = 0; i < 10; i++)
+            {
+                Thread.Sleep(100);
+                Console.Write(".");
+            }
+        });
+
+    start.Subscribe(
+        unit => Console.WriteLine("Unit published"), 
+        () => Console.WriteLine("Action completed"));
+}
+
+static void StartFunc()
+{
+    var start = Observable.Start(() =>
+    {
+        Console.Write("Working away");
+        for (int i = 0; i < 10; i++)
+        {
+            Thread.Sleep(100);
+            Console.Write(".");
+        }
+        return "Published value";
+    });
+
+    start.Subscribe(
+        Console.WriteLine, 
+        () => Console.WriteLine("Action completed"));
+}
+```
+
+Note the difference between `Observable.Start` and `Observable.Return`. The `Start` method invokes our callback only upon subscription, so it is an example of a 'lazy' operation. Conversely, `Return` requires us to supply the value up front.
+
+The observable returned by `Start` may seem to have a superficial resemblance to `Task` or `Task<T>` (depending on whether you use the `Action` or `Func<T>` overload). Each represents work that may take some time before eventually completing, perhaps producing a result. However, there's a significant difference: `Start` doesn't begin the work until you subscribe to it. Moreover, it will re-execute the callback every time you subscribe to it. So it is more like a factory for a task-like entity.
+
+### From events
+
+As we discussed early in the book, .NET has a model for events that is baked into its type system. This predates Rx (not least because Rx wasn't feasible until .NET got generics in .NET 2.0) so it's common for types to support events but not Rx. To be able to integrate with the existing event model, Rx provides methods to take an event and turn it into an observable sequence. I showed this briefly in the file system watcher example earlier, but let's examine this in a bit more detail. There are several different varieties you can use. This show the most succinct form:
+
+```cs
+FileSystemWatcher watcher = new (@"c:\temp");
```

Comment:
Definitely not, but I'm going to change the path because I evidently made the code suggest something I very much didn't intend it to.

I'm changing it to `c:\incoming` to get away from any suggestion of "temp folder". The reason I wrote `c:\temp` is just because it's where I tend to do experimental stuff—if I want to play around with `FileSystemWatcher` to understand how it works, I'd use the `c:\temp` folder. I do that specifically because that's **my** folder, and not a system folder. (That's why `System.IO.Path.GetTempPath` would definitely be the wrong thing. The problem with using the "real" temp folder is that all sorts of things use that, making it a problematic choice for experimentation.)

So in my head that signified "sandbox".

But I've switched to `c:\incoming` because that's not going to exist on most machines, but is suggestive of some sort of "drop" location that you might want to observe.